pkg/controller/releasemanager/incarnation_controller.go

```diff
@@ -50,11 +50,12 @@ func (c *ClusterInfoList) ExpectedReplicaCount(liveness bool, count int) int {
 }
 
 type IncarnationController struct {
-	deliveryClient client.Client
-	planApplier    plan.Applier
-	log            logr.Logger
-	releaseManager *picchuv1alpha1.ReleaseManager
-	clusterInfo    ClusterInfoList
+	deliveryClient  client.Client
+	deliveryApplier plan.Applier
```

Comment:
This was a good decision. I would have also started with just using the local client, but this is better.